src/wdio/services/compare/TerraCompare.js

```diff
@@ -0,0 +1,54 @@
+import BaseCompare from 'wdio-visual-regression-service/lib/methods/BaseCompare';
+import resemble from 'node-resemble-js';
+import get from 'lodash.get';
+
+export default class TerraCompare extends BaseCompare {
+  constructor(screenshotProcessor) {
+    super();
+    this.screenshotProcessor = screenshotProcessor;
```

Comment:
`screenshotProcessor` provides a mechanism for different implementations of getting reference screenshots, saving latest screenshots, saving diff screenshots, and handling a "no reference" screenshot.